### ISSUE TITLE
\Illuminate\Support\Facades\Auth vs \Auth

### ISSUE THREAD

**Issue by arxeiss** [nunomaduro/larastan#506]
- Larastan Version: 0.5.4
- `--level` used: 6

### Description
When I do `use Auth` I got errors, when changed to `\Illuminate\Support\Facades\Auth` errors dissappear.

### Laravel code where the issue was found

```php
use Auth;

$data['parent_id'] = Auth::user()->id;
```
Throws this error: `Access to an undefined property Illuminate\Contracts\Auth\Authenticatable::$id.`

```php
use Illuminate\Support\Facades\Auth;

$data['parent_id'] = Auth::user()->id;
```
This works perfectly

**Comment by szepeviktor** [comment]
Thank you for your report.
Using shortcuts like `use Auth` is not the nicest thing in the world.

**Comment by arxeiss** [comment]
FIY, I tried the following:

1. Removed `final` from class AuthExtension 
1. Created new Extension, which derives AuthExtension and only what I changed is

```php
public function getClass(): string
{
    return \Auth::class;
}
```

Then it is working, but I don't like solution of creating this to all aliases. Maybe this is just limitation and you can close this ticket. I already changed everything in my code from Auth to \Illuminate\Support\Facades\Auth

**Comment by szepeviktor** [comment]
> I already changed everything in my code from Auth to \Illuminate\Support\Facades\Auth

👍 

Should we say nasty things in this issue??? Using class aliases instead of typing the fully qualified class name may spare you some keystrokes but makes your imports cryptic.

**Comment by canvural** [comment]
Yeah, I agree with @szepeviktor And we can close this one if you changed everything to `\Illuminate\Support\Facades\Auth`

**Comment by mfn** [comment]
I wonder why the use of global vs. specific class fixed this for the user? 🤔

To me it's the same, whether I use `use Auth` or `\Illuminate\Support\Facades\Auth`, same outcome:
- `Access to an undefined property Illuminate\Contracts\Auth\Authenticatable::$id.`
  or
- `Parameter … of method … expects App\User, Illuminate\Contracts\Auth\Authenticatable|null given.`

I mean, the error is correct/clear 😄 `Authenticatable` contact != `App\User` model.

What's the recommended solution to avoid this error? `instanceof` checks? phpdoc?

**Comment by bernardwiesner** [comment]
@mfn  could you fix this? I am also getting same issue on 1.0.3

```php
Auth::user()->id;
````

Access to an undefined property Illuminate\Contracts\Auth\Authenticatable::$id.
I tried with both use Auth or \Illuminate\Support\Facades\Auth, same error...

Is there some workaround for this?

**Comment by mfn** [comment]
@bernardwiesner I don't remember much but I checked the commit in my private repo where I removed the `ignoreErrors` for the Facade issues and seems when I removed the `Auth`-Facade **alias** and switched over directly importing the `\Illuminate\Support\Facades\Auth` itself, it stopped being reported.

Interestingly I wrote before it did not matter:
> To me it's the same, whether I use use Auth or \Illuminate\Support\Facades\Auth, same outcome

But maybe something changed in between posting the above comment and my commit, which was July 2020 😏
